app/code/community/Twispay/Tpay/Helper/Data.php

```diff
@@ -1,8 +1,1097 @@
+      return FALSE;
+    }
+
+    /* Create the URL. */
+    $url = $url . '/order/' . $orderId;
+
+    /* Make the server request. */
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "GET");
+    curl_setopt($ch, CURLOPT_HTTPHEADER, ['accept: application/json', 'Authorization: ' . $apiKey]);
+    /* Send the request. */
+    $response = curl_exec($ch);
+    curl_close($ch);
+    /* Decode the response. */
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {
+      return $response->data->orderStatus;
+    } else {
+      Mage::Log(__FUNCTION__ . Mage::helper('tpay')->__(' Order details extraction failed: Server returned error: %s', $response->message), Zend_Log::ERR , $this->logFileName, /*forceLog*/TRUE);
+      return FALSE;
+    }
+  }
+
+
+  /**
+   * Function that cancels a recurring profile
+   *  in the Twispay platform.
+   * 
+   * @param profile: The recurring profile order for which to update the status.
+   * @param recurringProfileChildOrder: Child order of the recurring profile.
+   * @param message: The cancel reason.
+   */
+  public function cancelRecurringProfile($profile, $recurringProfileChildOrder, $message){
+    /* Get the config values. */
+    $apiKey = Mage::helper('tpay')->getApiKey();
+    Mage::Log(__FUNCTION__ . ': apiKey=' . print_r($apiKey, true), Zend_Log::DEBUG, $this->logFileName);
+    $url = Mage::helper('tpay')->getApiUrl();
+    Mage::Log(__FUNCTION__ . ': url=' . print_r($url, true), Zend_Log::DEBUG, $this->logFileName);
+
+    if ('' == $apiKey) {
+      Mage::Log(__FUNCTION__ . Mage::helper('tpay')->__(' Subscription cancel failed: Incomplete or missing configuration.'), Zend_Log::ERR , $this->logFileName, /*forceLog*/TRUE);
+      return;
+    }
+
+    /* Create the URL. */
+    $url = $url . '/order/' . $profile->getReferenceId();
+
+    /* Create the DELETE data arguments. */
+    $postData = 'message=' . $message;
+
+    /* Make the server request. */
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+    curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "DELETE");
+    curl_setopt($ch, CURLOPT_HTTPHEADER, ['accept: application/json', 'Authorization: ' . $apiKey]);
+    curl_setopt($ch, CURLOPT_POST, count($postData));
+    curl_setopt($ch, CURLOPT_POSTFIELDS, $postData);
+
+    /* Send the request. */
+    $response = curl_exec($ch);
+    curl_close($ch);
+    /* Decode the response. */
+    $response = json_decode($response);
+
+    /* Check if the response code is 200 and message is 'Success'. */
+    if ((200 == $response->code) && ('Success' == $response->message)) {

```

Comment:
We should also check if the response was decoded successfully with `json_decode`.

Reply:
Added the check.